python/cuML/knn/knn_wrapper.pyx

```diff
@@ -230,31 +261,68 @@ cdef class KNN:
         N = len(X)
 
         # Need to establish result matrices for indices (Nxk) and for distances (Nxk)
-        I_ndarr = cuda.to_device(np.zeros(N*k, dtype=np.int64))
-        D_ndarr = cuda.to_device(np.zeros(N*k, dtype=np.float32))
+        I_ndarr = cuda.to_device(np.zeros(N*k, dtype=np.int64, order = "F"))
+        D_ndarr = cuda.to_device(np.zeros(N*k, dtype=np.float32, order = "F"))
 
         cdef uintptr_t I_ptr = self._get_ctype_ptr(I_ndarr)
         cdef uintptr_t D_ptr = self._get_ctype_ptr(D_ndarr)
 
-        self.k.search(<float*>X_ctype,
-                      <int> N,
-                      <long*>I_ptr,
-                      <float*>D_ptr,
-                      <int> k)
+        self._query(X_ctype, N, k, I_ptr, D_ptr)
 
-        I_ndarr = I_ndarr.reshape((N, k)).transpose()
-        D_ndarr = D_ndarr.reshape((N, k)).transpose()
+        I_ndarr = I_ndarr.reshape((N, k))
+        D_ndarr = D_ndarr.reshape((N, k))
 
         I = cudf.DataFrame()
-        for i in range(0, I_ndarr.shape[0]):
-            I[str(i)] = I_ndarr[i,:]
+        for i in range(0, I_ndarr.shape[1]):
+            I[str(i)] = I_ndarr[:,i]
 
         D = cudf.DataFrame()
-        for i in range(0, D_ndarr.shape[0]):
-            D[str(i)] = D_ndarr[i,:]
+        for i in range(0, D_ndarr.shape[1]):
+            D[str(i)] = D_ndarr[:,i]
 
         if isinstance(X, np.ndarray):
             I = np.asarray(I.as_gpu_matrix())
             D = np.asarray(D.as_gpu_matrix())
 
         return D, I
+
+
+    def _query(self, X_ctype, N, k, I_ptr, D_ptr):
+        """
+        Query the KNN index for the k nearest neighbors of column vectors in X.
+
+        Parameters
+        ----------
+        X_ctype : Ctypes pointer (row-major)
+            Pointer to input data
+
+        N: Intetger
+            The number of rows in X
+
+        k: Integer
+            Number of neighbors to search
+
+        I_ptr: Ctypes pointer (row-major)
+            Pointer to N*k array for output indices
+
+        D_ptr: Ctypes pointer (row-major)
+            Pointer to N*k array for output distances
+
+        Returns
+        ----------
+        distances: cuDF DataFrame or numpy ndarray
+            The distances of the k-nearest neighbors for each column vector in X
+
+        indices: cuDF DataFrame of numpy ndarray
+            The indices of the k-nearest neighbors for each column vector in X
+        """
+
+        cdef uintptr_t I = I_ptr
```

Comment:
Had not thought about this but was just playing with adding flake8 rules for cython and this came up, `I` goes against the pep8 https://lintlyci.github.io/Flake8Rules/rules/E741.html, would be good to change the name.